mmocr/engine/hooks/visualization_hook.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os.path as osp
+from typing import Sequence, Union
+
+import mmcv
+from mmengine.hooks import Hook
+from mmengine.runner import Runner
+from mmengine.visualization import Visualizer
+
+from mmocr.registry import HOOKS
+from mmocr.structures import TextDetDataSample, TextRecogDataSample
+
+
+# TODO Files with the same name will be overwritten for multi datasets
+@HOOKS.register_module()
+class VisualizationHook(Hook):
+    """Detection Visualization Hook. Used to visualize validation and testing
+    process prediction results.
+
+    Args:
+        enable (bool): Whether to enable this hook. Defaults to False.
+        interval (int): The interval of visualization. Defaults to 50.
+        score_thr (float): The threshold to visualize the bboxes
+            and masks. It's only useful for text detection. Defaults to 0.3.
+        show (bool): Whether to display the drawn image. Defaults to False.
+        wait_time (float): The interval of show in seconds. Defaults
+            to 0.
+        file_client_args (dict): Arguments to instantiate a FileClient.
+            See :class:`mmcv.fileio.FileClient` for details.
+            Defaults to ``dict(backend='disk')``.
+    """
+
+    def __init__(
+        self,
+        enable: bool = False,
+        interval: int = 50,
+        score_thr: float = 0.3,
+        show: bool = False,
+        draw_pred: bool = False,
+        draw_gt: bool = False,
+        wait_time: float = 0.,
+        file_client_args: dict = dict(backend='disk')
+    ) -> None:
+        self._visualizer: Visualizer = Visualizer.get_current_instance()
+        self.interval = interval
+        self.score_thr = score_thr
+        self.show = show
+        self.draw_pred = draw_pred
+        self.draw_gt = draw_gt
+        self.wait_time = wait_time
+        self.file_client_args = file_client_args.copy()
+        self.file_client = None
+        self.enable = enable
+
+    def after_val_iter(self, runner: Runner, batch_idx: int,
```

Comment:
Looks the same as `after_test_iter`? We can extract them to a function and use it in `after_val_iter` and `after_test_iter`